File: src/main/scala/main/Food.scala
package main

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.Batch

import scala.util.Random

case class Food(cell: Cell)

object Food {
  def randomFood(snake: Snake): Food = {
    val x = Random.between(0, Config.widthInCells)
    val y = Random.between(0, Config.heightInCells)

    Food(Cell(x, y, new Color(1.0f, 0.0f, 0, 1)))
  }

  def draw(food: Food, batch: Batch, parentAlpha: Float): Unit = {
    Cell.draw(food.cell, batch, parentAlpha)
  }
}

File: src/main/scala/main/snake_oop/Food.java
package main.snake_oop;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;

public class Food {
    protected Cell cell;

    public Food() {
        cell = new Cell(0, 0, Color.RED);
        randomFood();
    }

    public void draw(Batch batch) {
        cell.draw(batch);
    }

    public void randomFood() {
        cell.x = getRandomNumber(0, 30);
        cell.y = getRandomNumber(0, 30);
    }

    public int getRandomNumber(int min, int max) {
        return (int) ((Math.random() * (max - min)) + min);
    }
}
